### ISSUE TITLE
Security Metric

### ISSUE THREAD

**Issue by abuhman** [chaoss/augur#24]
Update on Security Metric search:

I looked into cii-census project on Github.  It calculates and assigns a point value for security (more information here: https://www.coreinfrastructure.org/programs/census-project).  By reverse-engineering I found the data it gets does appear to be obtained manually (for example, through the command line) and from the Debian security tracker website, so I'm not sure we could follow the same procedures other than perhaps assigning some sort of point values.

They mention the following sources for their data:
"The data represented here is derived from: DSAs issued by the Security Team; issues tracked in the CVE database, issues tracked in the National Vulnerability Database (NVD), maintained by NIST; and security issues discovered in Debian packages as reported in the BTS."

As far as CVEs (Common Vulnerabilities and Exposures), what I have found when reading about them is that they appear to be voluntary to register.  In other words, more CVEs for a project may indicate acceptance of CVEs as the standard, greater ability to detect vulnerabilities, and/or more reporting rather than less security.  

I did find this site that gives a CVE count for various products https://www.cvedetails.com/product-list/firstchar-M/vendor_id-0/products.html .  As of now I'm not sure of a way to use it other than manually, though.

The National Vulnerability Database also keeps track of vulnerabilities.  I have not looked into it in depth.

In short, as of when I write this I have not found a clear way to evaluate security of a Github project or discover how many CVE's it has automatically.   

I do know that there are automated tools that can attack a running web application and create a results report based on any vulnerabilities, but that would require us manually installing and running our own copy of a web project (so, not viable).

**Comment by germonprez** [comment]
Perhaps take a look here: 

https://github.com/DoSOCSv2/DoSOCSv2

This is a project we built to interact directly with FOSSology scanners (licenses) and the NIST DB to get any reported CPEs. 

This could be a good start.

**Comment by abuhman** [comment]
Oops, beat me to it.  I mentioned that one in a separate issue.

**Comment by howderek** [comment]
You may want to look at [codeclimate](https://docs.codeclimate.com/docs/list-of-engines)'s work  because they have already successfully integrated the following static analysis tools: https://docs.codeclimate.com/docs/list-of-engines

A few of those engines look for security vulnerabilities in dependencies (nsp, Bundler-Audit)

**Comment by sgoggins** [comment]
A security metric seems like something to put on the road map.  Parking it for now.

**Comment by sgoggins** [comment]
@ccarterlandis and @ChristianCme : perhaps after we add additional visualizations to our release this week one or both of you could take a week and explore the implementation of security metrics, following the notes above. {Lets discuss and prioritize.}

**Comment by howderek** [comment]
Closing this due to inactivity
